agent.go

```diff
@@ -136,6 +144,7 @@ func (c *controller) handleKeyChange(keys []*types.EncryptionKey) error {
 			}
 		}
 	}
+	c.Unlock()
```

Comment:
Two concurrent go routines, one takes `c.Lock()` and tries `nDB.Lock()`; second go routine takes `nDB.Lock()` and tries `c.Lock()`. We had such deadlocks before, between network and endpoint locks IIRC.